Stable

danrings opened the pull request:
I couldn't manage to pass a string containing a "=" to sphinx via sphinx-build.py, which would give me much more flexibility configuring the build without having to create a new conf.py.
So this one-liner adds what I need.

Commit:
Allow the '=' character in the -D option of sphinx-build.py

Commit:
Allow the '=' character in the -D option of sphinx-build.py

tk0miya commented on sphinx/cmdline.py:
```diff
@@ -204,7 +204,8 @@ def main(argv):
         try:
-            key, val = val.split('=')
+            key, *vals = val.split('=')
+            val = '='.join(vals)
```
Did you mean `val.split('=', 1)`?

tk0miya commented:
LGTM with nits.
Could you also check why the tests failed?

Commit:
Make reading settings with option -D more robust in sphinx-build.py, by splitting only at first '='.

Commit:
Make reading settings with option -D more robust in sphinx-build.py, by splitting only at first '='.

danrings commented:
val.split('=', 1) seems to be okay except the documentation I read did not make it absolutely clear if a greedy or non-greedy split is performed, i.e. is the split done at the first or last '=' on the line? Anyways, I have changed it to use that command now.

 
Concerning the tests - it was down to the *vals syntax not present in V2.7 - not a problem anymore after my change. All tests pass now.
 
Thank you for your time,
Daniel

 
-----Ursprüngliche Nachricht-----
 Von: "Takeshi KOMIYA" [notifications@github.com]
 Gesendet: Do. 20.10.2016 08:06
 An: "sphinx-doc/sphinx" [sphinx@noreply.github.com]
 Kopie: "Daniel" [topwohnung@freenet.de],  "Author" [author@noreply.github.com]
 Betreff: Re: [sphinx-doc/sphinx] Stable (#3068)

@tk0miya commented on this pull request.
In sphinx/cmdline.py:

> @@ -204,7 +204,8 @@ def main(argv):confoverrides = {}for val in opts.define:try:-            key, val = val.split('=')+            key, *vals = val.split('=')+            val = '='.join(vals)Did you mean val.split('=', 1)?
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> -----Ursprüngliche Nachricht Ende-----

---

Die Bundesliga hat begonnen! Alle Tore, alle Ergebnisse, alle News: Pocket Liga jetzt im AppStore oder Google Play Store herunterladen - kostenlos!

tk0miya commented:
Merged!
Thank you for your contribution.
